pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -268,13 +272,28 @@ protected GenericRow doUpdateRecord(GenericRow record, RecordInfo recordInfo) {
             int currentDocId = recordLocation.getDocId();
             if (currentQueryableDocIds == null || currentQueryableDocIds.contains(currentDocId)) {
               _reuse.clear();
-              previousRecordReference.set(currentSegment.getRecord(currentDocId, _reuse));
+              for (String column: record.getFieldToValueMap().keySet()) {
+                PinotSegmentColumnReader pinotSegmentColumnReader =
+                    new PinotSegmentColumnReader(recordLocation.getSegment(), column);
```

Comment:
We should create `PinotSegmentColumnReader` only when we actually need to read the value. In override mode, if the value exist, we don't need to create it

Reply:
Also we need to close it after reading the value

Reply:
gotcha, thanks